server/workflow_template_server.go

```diff
@@ -0,0 +1,334 @@
+		return nil, err
+	}
+
+	workflowTemplateVersions, err := client.ListWorkflowTemplateVersions(req.Namespace, req.Uid)
+	if err != nil {
+		return nil, err
+	}
+
+	var workflowTemplates []*api.WorkflowTemplate
+	for _, wtv := range workflowTemplateVersions {
+		workflowTemplates = append(workflowTemplates, apiWorkflowTemplate(wtv))
+	}
+
+	return &api.ListWorkflowTemplateVersionsResponse{
+		Count:             int32(len(workflowTemplateVersions)),
+		WorkflowTemplates: workflowTemplates,
+	}, nil
+}
+
+func (s *WorkflowTemplateServer) ListWorkflowTemplates(ctx context.Context, req *api.ListWorkflowTemplatesRequest) (*api.ListWorkflowTemplatesResponse, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "list", "argoproj.io", "workflows", "")
+	if err != nil || !allowed {
+		return nil, err
+	}
+
+	workflowTemplates, err := client.ListWorkflowTemplates(req.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	apiWorkflowTemplates := []*api.WorkflowTemplate{}
+	for _, wtv := range workflowTemplates {
+		apiWorkflowTemplates = append(apiWorkflowTemplates, apiWorkflowTemplate(wtv))
+	}
+
+	return &api.ListWorkflowTemplatesResponse{
+		Count:             int32(len(apiWorkflowTemplates)),
+		WorkflowTemplates: apiWorkflowTemplates,
+	}, nil
+}
+
+func (s *WorkflowTemplateServer) ArchiveWorkflowTemplate(ctx context.Context, req *api.ArchiveWorkflowTemplateRequest) (*api.ArchiveWorkflowTemplateResponse, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "delete", "argoproj.io", "workflows", "")
+	if err != nil || !allowed {
+		return nil, err
+	}
+
+	archived, err := client.ArchiveWorkflowTemplate(req.Namespace, req.Uid)
+	if err != nil {
+		return nil, err
+	}
+
+	return &api.ArchiveWorkflowTemplateResponse{
+		WorkflowTemplate: &api.WorkflowTemplate{
+			IsArchived: archived,
+		},
+	}, nil
+}
+
+func (s *WorkflowTemplateServer) GetWorkflowTemplateLabels(ctx context.Context, req *api.GetWorkflowTemplateLabelsRequest) (*api.GetLabelsResponse, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "create", "argoproj.io", "workflows", "")
+	if err != nil || !allowed {
+		return nil, err
+	}
+
+	labels, err := client.GetWorkflowTemplateLabels(req.Namespace, req.Name, label.TagPrefix, req.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &api.GetLabelsResponse{
+		Labels: mapToKeyValue(labels),
+	}
+
+	return resp, nil
+}
+
+// Adds any labels that are not yet associated to the workflow execution.
+// If the label already exists, overwrites it.
+func (s *WorkflowTemplateServer) AddWorkflowTemplateLabels(ctx context.Context, req *api.AddLabelsRequest) (*api.GetLabelsResponse, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "create", "argoproj.io", "workflows", "")
```

Comment:
I believe the verb here should be `"update"`

Reply:
Fixed.